ui/src/main/resources/edu/wpi/grip/ui/templates/Pipeline.vm

```diff
@@ -0,0 +1,100 @@
+#**
+This creates the complete java pipeline.
+*#
+#set($needsMut = false)
+#set($interpolation = false)
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.HashMap;
+import javax.imageio.ImageIO;
+import javax.swing.ImageIcon;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+
+import org.opencv.core.*;
+import org.opencv.core.Core.*;
+import org.opencv.imgcodecs.Imgcodecs;
+import org.opencv.imgproc.*;
+
+/**
+* Pipeline class.
+*
+* <P>Autogenerated pipeline from GRIP.
+*
+* <P>Make sure to set all sources using the setters before running processImage().
+*
+* <P>Tutorials and examples can be found online
+*
+* @author GRIP
+*/
+public class Pipeline{
+
+	//This map links the Outputs with their names
+	protected Map<String,Object> outputs;
+#foreach($source in $pipeline.getSources())
+    protected ${source.type()} $source.value();
+#end
+
+	/**
+	 * This constructor sets up the pipeline
+	 */
+	public Pipeline(){
+		outputs = new HashMap<String,Object>();
+	}
+
+	/**
+	 * This is the primary method that runs the entire pipeline and updates the outputs.
+	 */
+	public void processImage(){
```

Comment:
This should use a lock. Otherwise one thread could be grabbing outputs while another thread is running the pipeline.